packages/commons/src/interop-token/models.ts

```diff
@@ -3,134 +3,255 @@ import {
+    // z.discriminatedUnion performs better than z.union and gives more meaningful parsing errors.
+    role: z.undefined(),
+    "user-roles": CommaSeparatedStringToArray(UserRole),
+    uid: z.string().uuid(),
+    organizationId: z.string().uuid(),
+    selfcareId: z.string().uuid(),
+    organization: z.object({
+      id: z.string().uuid(),
+      name: z.string(),
+      roles: z.array(
+        z.object({
+          partyRole: z.string().nullish(),
+          role: UserRole,
+        })
+      ),
+      fiscal_code: z.string().nullish(),
+      ipaCode: z.string().nullish(),
+    }),
+    externalId: z.object({
+      origin: z.string(),
+      value: z.string(),
+    }),
+    name: z.string().nullish(),
+    family_name: z.string().nullish(),
+    email: z.string().nullish(),
+  })
+);
+export type InteropJwtUIPayload = z.infer<typeof InteropJwtUIPayload>;
+
+// ==========================================
+//    Interop INTERNAL Token
+// ==========================================
+export const InteropJwtInternalPayload = InteropJwtCommonPayload.merge(
+  z.object({
+    sub: z.string(),
+    role: z.literal(systemRole.INTERNAL_ROLE),
+  })
+);
+export type InteropJwtInternalPayload = z.infer<
+  typeof InteropJwtInternalPayload
+>;
 
 export type InteropInternalToken = {
   header: InteropJwtHeader;
   payload: InteropJwtInternalPayload;
   serialized: string;
 };
 
-/* ========================================== 
-    Interop SESSION Token 
-  ========================================== */
-const Organization = z.object({
-  id: z.string(),
-  name: z.string(),
-  roles: z.array(z.object({ role: z.string() })),
-});
+// ==========================================
+//    Interop SESSION Token
+// ==========================================
 export const SessionClaims = z.object({
-  [UID]: z.string(),
-  [ORGANIZATION]: Organization,
-  [NAME]: z.string().optional(),
-  [FAMILY_NAME]: z.string().optional(),
-  [EMAIL]: z.string().optional(),
+  uid: z.string(),
+  organization: z.object({
+    id: z.string(),
+    name: z.string(),
+    roles: z.array(z.object({ role: z.string() })),
+  }),
+  name: z.string().optional(),
+  family_name: z.string().optional(),
+  email: z.string().optional(),
 });
 export type SessionClaims = z.infer<typeof SessionClaims>;
 
-export const CustomClaims = z.object({
-  [USER_ROLES]: z.string(),
-  [ORGANIZATION_ID_CLAIM]: z.string(),
-  [SELFCARE_ID_CLAIM]: z.string(),
-  [ORGANIZATION_EXTERNAL_ID_CLAIM]: z.object({
-    [ORGANIZATION_EXTERNAL_ID_ORIGIN_CLAIM]: z.string(),
-    [ORGANIZATION_EXTERNAL_ID_VALUE_CLAIM]: z.string(),
+export const InteropUserJwtPayload = z.object({
+  "user-roles": z.string(),
+  organizationId: z.string(),
+  selfcareId: z.string(),
+  externalId: z.object({
+    origin: z.string(),
+    value: z.string(),
   }),
 });
-export type CustomClaims = z.infer<typeof CustomClaims>;
+export type InteropUserJwtPayload = z.infer<typeof InteropUserJwtPayload>;
```

Comment:
Session token seems to be a UI token, can't we delete these and just reuse `InteropJwtUIPayload`?